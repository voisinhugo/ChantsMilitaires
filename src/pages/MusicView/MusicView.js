import React from 'react';
import {Text, ScrollView, StyleSheet, View, SafeAreaView} from 'react-native';
import songs from '../../data';
import theme from '../../theme';

const styles = StyleSheet.create({
  container: {
    flexGrow: 1,
    backgroundColor: theme.color.background,
  },
  lyrics: {
    margin: theme.margin.x2,
    fontSize: theme.fontSize.body,
    color: theme.color.white,
  },
});

export const MusicView = ({navigation}) => {
  const musicId = navigation.getParam('musicId');

  return (
    <View style={styles.container}>
      <SafeAreaView>
        <ScrollView>
          <Text style={styles.lyrics}>{songs[musicId].lyrics}</Text>
        </ScrollView>
      </SafeAreaView>
    </View>
  );
};
