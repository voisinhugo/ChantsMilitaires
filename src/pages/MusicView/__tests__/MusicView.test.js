import React from 'react';
import {render} from 'react-native-testing-library';
import {MusicView} from '../MusicView';
import {getPropsWithNavigation} from '../../../lib/tests/helpers';

const song0Id = 0;
jest.mock('../../../data/index', () => ({
  '0': {
    id: 0,
    title: 'toto',
    lyrics:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed non risus. Suspendisse lectus tortor, dignissim sit amet, adipiscing nec, ultricies sed, dolor.',
  },
  '1': {id: 1, title: 'tata', lyrics: ''},
}));

const getParam = param => {
  const navigationParams = {musicId: song0Id};
  return navigationParams[param];
};
const {navigation} = getPropsWithNavigation({}, {getParam});

describe('MusicView', () => {
  it('should render', () => {
    const musicView = render(<MusicView navigation={navigation} />);

    expect(musicView).toMatchSnapshot();
  });
});
