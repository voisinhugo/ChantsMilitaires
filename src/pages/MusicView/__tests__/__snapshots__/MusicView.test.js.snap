// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MusicView should render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#272727",
      "flexGrow": 1,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
  >
    <RCTScrollView>
      <View>
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 16,
              "margin": 16,
            }
          }
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed non risus. Suspendisse lectus tortor, dignissim sit amet, adipiscing nec, ultricies sed, dolor.
        </Text>
      </View>
    </RCTScrollView>
  </RCTSafeAreaView>
</View>
`;
